tuna/fin_eval.py

```diff
@@ -179,45 +180,51 @@ def get_fin_input(self):
+            'arch: %s, num_cu: %s, direction: %s',
+            self.config.id, self.solver_id_map[fdb_obj['solver_name']],
+            self.dbt.session.arch, self.dbt.session.num_cu, self.config.direction)
+        raise ValueError("Unable to query find db entry")
+      fdb_entry = obj
+      fdb_entry.alg_lib = fdb_obj['algorithm']
+      fdb_entry.kernel_time = fdb_obj['time']
+      fdb_entry.workspace_sz = fdb_obj['workspace']
+      fdb_entry.session = self.dbt.session.id
+      fdb_entry.params = fdb_obj['params']
+    else:
+      self.logger.warning("Not evaluated: job(%s), solver(%s), %s", self.job.id,
+                          fdb_obj['solver_name'], fdb_obj['reason'])
+
+    self.logger.info('Updating find db(Eval) for job_id=%s', self.job.id)
+    session.commit()
+
+    return True
+
   def process_fdb_eval(self, fin_json, result_str='miopen_find_eval_result'):
     """process find db eval json results"""
     status = []
-    for fdb_obj in fin_json[result_str]:
-      self.logger.info('Processing object: %s', fdb_obj)
-      slv_stat = get_fin_slv_status(fdb_obj, 'evaluated')
-      status.append(slv_stat)
-      with DbSession() as session:
-        if fdb_obj['evaluated']:
-          obj, _ = self.get_fdb_entry(
-              session, self.solver_id_map[fdb_obj['solver_name']])
-          if not obj:
-            self.logger.info(
-                'Unable to find fdb entry for config: %s, solver: %s, '\
-                'arch: %s, num_cu: %s, direction: %s',
-                self.config.id, self.solver_id_map[fdb_obj['solver_name']],
-                self.dbt.session.arch, self.dbt.session.num_cu, self.config.direction)
-            raise ValueError("Unable to query find db entry")
-          fdb_entry = obj
-          fdb_entry.alg_lib = fdb_obj['algorithm']
-          fdb_entry.kernel_time = fdb_obj['time']
-          fdb_entry.workspace_sz = fdb_obj['workspace']
-          fdb_entry.session = self.dbt.session.id
-          fdb_entry.params = fdb_obj['params']
-        else:
-          self.logger.warning("Not evaluated: job(%s), solver(%s), %s",
-                              self.job.id, fdb_obj['solver_name'],
-                              fdb_obj['reason'])
-
-        self.logger.info('Updating find db(Eval) for job_id=%s', self.job.id)
-        try:
-          session.commit()
-        except OperationalError as err:
-          self.logger.warning('FinEval: Unable to update Database: %s', err)
-          status = [{
-              'solver': 'all',
-              'success': False,
-              'result': 'FinEval: Unable to update Database: {}'.format(err)
-          }]
+    fdb_obj = None
+    with DbSession() as session:
+      for fdb_obj in fin_json[result_str]:
+        self.logger.info('Processing object: %s', fdb_obj)
+        slv_stat = get_fin_slv_status(fdb_obj, 'evaluated')
+        callback = self.update_fdb_eval_entry
+        actuator = lambda x: x(session, fdb_obj)
```

Comment:
[Cosmetic] Perhaps these variables are not required, since the same may be passed in  directly.

Reply:
changed